sound/soc/sof/intel/cnl.c

```diff
@@ -233,12 +234,26 @@ void cnl_ipc_dump(struct snd_sof_dev *sdev)
 
 static int cnl_register_clients(struct snd_sof_dev *sdev)
 {
-	return intel_register_ipc_test_clients(sdev);
+	int ret;
+
+	ret = intel_register_ipc_test_clients(sdev);
+	if (ret < 0)
+		return ret;
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_PROBES)
+	return sof_client_dev_register(sdev, "probes", 0);
+#endif
+
+	return 0;
```

Comment:
so if PROBES are enabled, you get two `return` statements one after another. You aren't getting warnings something about unreachable code? The easiest way to improve this seems to me
```
#if ...
ret = sof_client_dev_register(...);
#endif

return ret;
```
but you can come up with something better. on all platforms of course.

Reply:
good point also but interestingly I don't have any warning from the compiler.

Reply:
I would still change this